src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,48 @@
+/// <reference types="vite-plugin-svgr/client" />
+import Check from '../../assets/icons/check.svg?react';
+import { Minus } from '@phosphor-icons/react';
+
+interface CheckboxComponentProps {
+  checked?: boolean;
+  indeterminate?: boolean;
+  disabled?: boolean;
+  onClick?: (e: unknown) => void;
+  required?: boolean;
+}
+
+const Checkbox = ({
+  checked = true,
+  indeterminate = false,
+  disabled = false,
+  onClick,
+  required,
+}: CheckboxComponentProps): JSX.Element => {
+  return (
+    <>
+      <button
+        onClick={onClick}
+        onKeyDown={() => {}}
```

Comment:
Should we have a function like onClick for this? Just in case

Reply:
Done